cli/main.go

```diff
@@ -35,20 +35,23 @@ func executeRootCmdWithContext() error {
 }
 
 func main() {
+	exitCode := 0
 	defer func() {
 		err := recover()
 		if err != nil {
 			originalMessage := fmt.Sprintf("panic: %v\n\n%s", err, string(debug.Stack()))
 			sentry.CurrentHub().CaptureMessage(originalMessage)
 			panic(err)
 		}
+
+		os.Exit(exitCode)
 	}()
+	defer cmd.CloseLogFile()
 
 	// This ensures we don't print anything until logging is configured
 	log.Logger = log.Level(zerolog.Disabled)
 	if err := executeRootCmdWithContext(); err != nil {
 		log.Error().Err(err).Msg("exiting with error")
-		//nolint:all This is fine if deferred is not called because there was no panic
-		os.Exit(1)
```

Comment:
`os.Exit` skips any deferred functions hence if we want to close the log file we need to defer `os.Exit` too

Reply:
[runtime.Goexit()](https://pkg.go.dev/runtime#Goexit)?

Reply:
@candiduslynx can you explain:

```
Calling Goexit from the main goroutine terminates that goroutine without func main returning. Since func main has not returned, the program continues execution of other goroutines. If all other goroutines exit, the program crashes.
```